### ISSUE TITLE
Is this still relevant?

### ISSUE THREAD

**Issue by Atemu** [microg/android_packages_apps_GsfProxy#22]
Hi, I'm wondering whether GsfProxy is still necessary in modern Android because it hasn't seen a release in years.

The docs aren't entirely clear on this and generally appear quite outdated.

**Comment by david7919** [comment]
Well, it says gsfproxy is **legacy** support proxy library so I guess its for older android versions or apps that use old code that needs this

**Comment by Atemu** [comment]
I'd like some confirmation on exactly that.

**Comment by david7919** [comment]
So, dev said this:
"It is used only by old apps, so it is still useful but not in all cases."

https://github.com/microg/GmsCore/issues/2623#issuecomment-2473637466

**Comment by Atemu** [comment]
~~@Sapiosenses is not a µG dev, though they appear to be knowledgeable.~~

Edit: You were talking about https://github.com/microg/GmsCore/issues/2623#issuecomment-2473637466.

**Comment by Sapiosenses** [comment]
> So, dev said this: "It is used only by old apps, so it is still useful but not in all cases."
> 
> [microg/GmsCore#2623 (comment)](https://github.com/microg/GmsCore/issues/2623#issuecomment-2473637466)


Ask yourself:

That service was shutdown by Google almost 10 years ago. That means that no app today that continues to use that old push messaging API will work on **any** stock Android device.

Furthermore, since that is a GMS service, it will not work on any plain vanilla AOSP-based ROM which was last updated in the last 10+ years, because that functionality would have been removed from AOSP by that time already. (Google moved it to proprietary GMS/Gapps frameworks, like many other things over the years)

Literally the only way ANY app can still use that service, is on a custom ROM running microG. (Which just picks up the communication attempts and forwards them to the current push messaging service)

So then one has to ask themselves, how many apps are there still out there, written in the last ~10 years, that have specifically coded themselves to only have full functionality on a microG device for push messages?

Apps like that were probably written within the first 4 years of Android's existence, if that.

How many of those do you think people are still using??

I'd guess the number is infinitesimally small.

In short: the impact of that being entirely missing will probably go unnoticed by 99.5% of users.

If @ale5000-git thinks it would be easy to update it for A15 that's cool. But I'd be shocked if anyone actually noticed the change outside of the error message that Google added for apps targeting those old APIs on A13-A15 devices.

**Comment by ale5000-git** [comment]
I don't think this app does have any problem with Android 15, if you install with adb I think you can bypass the sdk limit.

Beside supporting old apps it is also useful to fake Google since it make sure there is an app called `com.google.android.gsf` installed.
Although it isn't usually checked, apps may choose to detect this.

**Comment by ale5000-git** [comment]
To see an app that use it, look here: #17

**Comment by CircuitSerialKiller** [comment]
@Atemu @ale5000-git  
I apologize for the late reply. I can confirm GsfProxy is still used as discussed in #17 and necessary for registration with the latest version of the app. 
Also, would it be possible for someone to update the release here:
https://github.com/microg/GsfProxy/releases/tag/v0.1.0
with the compiled & signed version with the forcequeryable attribute set?

**Comment by ale5000-git** [comment]
> [@Atemu](https://github.com/Atemu) [@ale5000-git](https://github.com/ale5000-git) I apologize for the late reply. I can confirm GsfProxy is still used as discussed in [#17](https://github.com/microg/GsfProxy/issues/17) and necessary for registration with the latest version of the app. Also, would it be possible for someone to update the release here: [`v0.1.0` (release)](https://github.com/microg/GsfProxy/releases/tag/v0.1.0) with the compiled & signed version with the forcequeryable attribute set?

Currently there isn't a new official version, but there is a version signed by me here: https://github.com/micro5k/microg-unofficial-installer/blob/main/zip-content/origin/priv-app/GsfProxyA5K.apk
